### ISSUE TITLE
autoconf pcre++ check is not valid

### ISSUE THREAD

**Issue by simsong** [google/aff4#16]
This is in configure.ac:

```
AC_CHECK_LIB([pcre++], [main], [], [AC_MSG_ERROR([pcre++ library (libpcre++-dev) not found])])
```

My understanding is that `[main]` means that a function named `main()` needs to be present. However, I have the pcre++ package installed (via ports, with `port install pcrexx`) and this test is still failing. 

Here is a `AC_CHECK_LIB` from another project:

```
AC_CHECK_LIB([dl],[dlopen])
```

In this example, the `AC_CHECK_LIB` is making sure that a function called `dlopen` is present in the `dl` library.

It's unclearly why the check is checking for `main`

I'm not sure if this works on Linux, but it doesn't work on MacOS and it may not work on mingw.

**Comment by blschatz** [comment]
Did invoke configure per the instructions in the INSTALL file? The OSX specific instructions there add some extra include/libdir paths that might be a factor.

**Comment by simsong** [comment]
No, I just ran the autogen.sh and then ./configure. If that doesn't work, then it should be made to work.  Computers don't read README files...

**Comment by blschatz** [comment]
Agreed.
